datalad/distribution/add.py

```diff
@@ -336,17 +345,34 @@ def __call__(
                           'note': "no annex at %s" % ds.path}
                          for f in calls[dspath]['addurl_f']]
                     )
+            return_values = None  # to avoid mis-use
+
+        # XXX or we could return entire datasets_return_values, could be useful
+        # that way.  But then should be unified with the rest of commands, e.g.
+        # get etc
+        return_values_flat = []
+        for dspath, return_values in datasets_return_values.items():
+            if save and len(return_values):
+                # we got something added -> save
+                # everything we care about at this point should be staged already
+                Save.__call__(
+                    message='[DATALAD] added content',
+                    dataset=ds,
+                    auto_add_changes=False,
+                    recursive=False)
+            # TODO: you feels that this is some common logic we already have somewhere
+            dsrelpath = relpath(dspath, dataset.path)
```

Comment:
I share your feeling :)

At least it's used in `get` already (line 156 and following). But basically this is needed whenever we recurse and want to report file paths.